### PR TITLE
auth only available in login command

### PR DESCRIPTION
**Pull request by 0xbase12**
Connected to #372

### CODE REVIEW

File: client/src/main/python/slipstream/SlipStreamHttpClient.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def _assemble_endpoints(self):
         self.runEndpoint = self.serviceurl + util.RUN_RESOURCE_PATH
         self.run_url = self.runEndpoint + '/' + self.diid
 
-        self.authnServiceUrl = self.serviceurl + '/'
+        self.authnServiceUrl = self.serviceurl + '/api/session'
```
Comment (loomis) [inline client/src/main/python/slipstream/SlipStreamHttpClient.py]: The `auth/login` URL is deprecated.  The session resource on the CIMI server should be used instead.

File: client/src/main/python/slipstream/SlipStreamHttpClient.py [hunk 2]
```diff
@@ -286,7 +286,8 @@ def get_server_configuration(self):
         return config
 
     def login(self, username, password):
-        self._httpPost(self.serviceurl + '/auth/login', body={
+        self._httpPost(self.authnServiceUrl, body={
+            'href': 'session-template/internal',
```
Comment (loomis) [inline client/src/main/python/slipstream/SlipStreamHttpClient.py]: This should be extended to allow any authentication method to be used.  The form needs to have the key "href" and then any keys defined with the referenced template.  For "internal", the keys are "username" and "password"; for "api-key", "key" and "secret".

Comment (loomis) [reply client/src/main/python/slipstream/SlipStreamHttpClient.py]: (Not necessarily with this PR.)

File: client/src/test/python/TestHttpClient.py [hunk 3]
```diff
@@ -50,7 +50,8 @@ def test_get_with_username_password(self):
 
         args, kwargs = client.session.request.call_args
         self.assertEqual(('GET', 'http://foo.bar'), args)
-        self.assertTrue(kwargs['auth'], ('username', 'password'))
+        # What's the correct behavior here?
+        #self.assertTrue(kwargs['auth'], ('username', 'password'))
```
Comment (loomis) [inline client/src/test/python/TestHttpClient.py]: I had to comment this out to get the test to pass.  What's the expected behavior here now?  Is this test even useful now?

File: client/src/main/python/slipstream/SlipStreamHttpClient.py (hunk 2 above)
Comment (0xbase12) [reply client/src/main/python/slipstream/SlipStreamHttpClient.py]: Yes I'm agree with you.

File: client/src/test/python/TestHttpClient.py (hunk 3 above)
Comment (0xbase12) [reply client/src/test/python/TestHttpClient.py]: Thanks, this test is no more useful.

**Comment by loomis** [comment]
I think that more of the uses of the HTTPClient need to be fixed.  Here's a stacktrace from the `ss-module-get` command:
```
pypi> ss-module-get apps
Traceback (most recent call last):
  File "/Users/loomis/Library/Python/2.7/bin/ss-module-get", line 73, in <module>
    MainProgram()
  File "/Users/loomis/Library/Python/2.7/bin/ss-module-get", line 37, in __init__
    super(MainProgram, self).__init__(argv)
  File "/Users/loomis/Library/Python/2.7/lib/python/site-packages/slipstream/command/CommandBase.py", line 89, in __init__
    self._callAndHandleErrorsForCommands(self.doWork.__name__)
  File "/Users/loomis/Library/Python/2.7/lib/python/site-packages/slipstream/command/CommandBase.py", line 122, in _callAndHandleErrorsForCommands
    res = self.__class__.__dict__[methodName](self, *args, **kw)
  File "/Users/loomis/Library/Python/2.7/bin/ss-module-get", line 59, in doWork
    client = HttpClient(self.options.username, self.options.password)
AttributeError: Values instance has no attribute 'username'
```

**Comment by loomis** [comment]
Just removing the username/password references seems to work (adfa9f7), but there may be more subtle changes needed.

**Pull request merged**